core/src/main/java/org/pentaho/platform/web/http/api/resources/SchedulerResource.java

```diff
@@ -928,15 +928,89 @@ public void removeJob( IJobRequest jobRequest ) {
   } )
   public Response deleteJob( JobRequest jobRequest ) {
     try {
-      if ( schedulerService.removeJob( jobRequest.getJobId() ) ) {
-        return buildPlainTextOkResponse( "REMOVED" );
-      }
-      return buildPlainTextOkResponse( schedulerService.getJob( jobRequest.getJobId() ).getState().name() );
+      return buildPlainTextOkResponse( deleteJob( jobRequest.getJobId() ) );
     } catch ( SchedulerException e ) {
       throw new RuntimeException( e );
     }
   }
 
+  private String deleteJob( String jobId ) throws SchedulerException {
+    if ( schedulerService.removeJob( jobId ) ) {
+      return REMOVED_JOB_STATE;
+    }
+
+    return schedulerService.getJob( jobId ).getState().name();
+  }
+
+  /**
+   * Deletes all the specified scheduled jobs from the platform.
+   *
+   * <p><b>Example Request:</b><br />
+   * DELETE pentaho/api/scheduler/removeJobs
+   * </p>
+   * <br /><b>DELETE data:</b>
+   * <pre function="syntax.json">
+   *   &lt;jobsRequest&gt;
+   *     &lt;jobIds&gt;
+   *       &lt;jobId&gt;admin  BlockoutAction 1410786491503&lt;/jobId&gt;
+   *       &lt;jobId&gt;admin  BlockoutAction 1410786491503&lt;/jobId&gt;
+   *     &lt;/jobIds&gt;
+   *   &lt;/jobsRequest&gt;
+   * </pre>
+   *
+   * <p><b>Example Response:</b></p>
+   * <pre function="syntax.xml">
+   *   &lt;JobsResponse&gt;
+   *     &lt;changes&gt;
+   *       &lt;entry&gt;
+   *         &lt;key&gt;admin  BlockoutAction 1410786491503&lt;/key&gt;
+   *         &lt;value&gt;REMOVED&lt;/value&gt;
+   *       &lt;/entry&gt;
+   *       &lt;entry&gt;
+   *         &lt;key&gt;admin  BlockoutAction 1410786491503&lt;/key&gt;
+   *         &lt;value&gt;ERROR&lt;/value&gt;
+   *       &lt;/entry&gt;
+   *     &lt;/changes&gt;
+   *   &lt;/JobsResponse&gt;
+   * </pre>
+   *
+   * @param jobsRequest A JobsRequest object containing a list of jobIds.
+   * @return A jax-rs Response object containing all the scheduled jobs ids and their new status.
+   */
+  @DELETE
```

Comment:
Usually DELETE is used with the resource identified in the URI (e.g. DELETE /job/1), are we sure there will be no problems with including a message body in it?
According to [RFC 9110](https://www.rfc-editor.org/rfc/rfc9110#section-9.3.5-6) (June 2022)

> Although request message framing is independent of the method used, content received in a DELETE request has no generally defined semantics, cannot alter the meaning or target of the request, and might lead some implementations to reject the request and close the connection because of its potential as a request smuggling attack ([Section 11.2](https://www.rfc-editor.org/rfc/rfc9112#section-11.2) of [[HTTP/1.1](https://www.rfc-editor.org/rfc/rfc9110#HTTP11)]). A client SHOULD NOT generate content in a DELETE request unless it is made directly to an origin server that has previously indicated, in or out of band, that such a request has a purpose and will be adequately supported. An origin server SHOULD NOT rely on private agreements to receive content, since participants in HTTP communication are often unaware of intermediaries along the request chain.

There seems to have been some cases were clients are unable to include the message body, and others where proxies/firewalls strip them from the DELETE request.

Have you considered using a POST?

Reply:
I have not considered it, but I was in doubt when I created the endpoint because this is not standard REST (the bulk action). I will implement your suggestion, because I think it covers better all scenarios.